osquery/tables/networking/curl_certificate.cpp

```diff
@@ -386,27 +487,49 @@ Status getTLSCertificate(const std::string& hostname,
   auto cert = std::unique_ptr<X509, decltype(delX509)>(
       SSL_get_peer_certificate(ssl), delX509);
   if (cert == nullptr) {
+#ifdef WIN32
+    WSACleanup();
+#endif
     return Status::failure("No certificate");
   }
 
   Row r;
   r["hostname"] = hostname;
-  fillRow(r, cert.get(), dump_certificate);
+  fillRow(r, cert.get(), dump_certificate, timeout);
   results.push_back(r);
+#ifdef WIN32
+  WSACleanup();
+#endif
   return Status::success();
 }
 
 QueryData genTLSCertificate(QueryContext& context) {
   QueryData results;
   auto dump_certificate = 0;
+  auto timeout = DEFAULT_READ_TIMEOUT;
   auto hostnames = context.constraints["hostname"].getAll(EQUALS);
 
   if (context.hasConstraint("dump_certificate", EQUALS)) {
     dump_certificate = context.constraints["dump_certificate"].matches<int>(1);
   }
 
+  if (context.hasConstraint("timeout", EQUALS)) {
+    auto timeout_str = *(context.constraints["timeout"].getAll(EQUALS).begin());
+    try {
```

Comment:
Instead of the `try` you can use our utility method `tryTo`, for example
```cpp
auto timeout = tryTo<int>(timeout_str, 10).takeOr(DEFAULT_READ_TIMEOUT);
```